### DISCUSSION TITLE
Dudas sobre como trabajar con git

### DISCUSSION THREAD

**Discussion by Carlo-MG** [community/community#128068] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Tengo dudas sobre cómo hacer un push, commit y la creación y administración de ramas. Sobre los push, la duda que tengo es cuándo debo hacer un push a un repositorio, es decir, si debo hacerlo a partir de cierta cantidad de commits, cuando ya terminé las funcionalidades de una rama o en qué punto del proyecto hacerlo. En cuanto a los commits, la duda que tengo es qué tipo de mensajes debo poner y cuándo debo hacerlos, ya que tengo el hábito de hacer commit aunque los cambios que haga sean tan insignificantes que no sé qué mensajes poner. Por último, sobre las ramas, me gustaría saber qué nombres debería usar.

**Comment by davevad93** [comment]
Hi @Carlo-MG ,

About pushing your code to a remote repo in my opinon the less `commits` the better, because if you do a `push` with many commits there's a high risk of conflicts that can be difficult to resolve. Personally I make a `git push` for each `commit`, two at most.

About commits messages, they should be as descriptive as possible without being too long or out of context, the typical structure of `git commit` messages is the following: `<type>(<scope>): <subject>`.
You should only commit when a logical component is completed and you've checked that everything works, for example make a commit when you've finished to implement a function in your code, not when you've just created a bunch of variables.

About naming conventions for branches, they should be lowercase and reflect their purpose or task, for instance if you want to fix something create a new `branch` called `bugfix`. 

I hope you find it helpful!

**Comment by Ahmed-Alaini** [comment]
Responce 1
